ui/app/components/secret-list-header-tab.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * @module SecretListHeaderTab
+ * SecretListHeaderTab component passes in properties that are used to check capabilities and either display or not display the component.
+ * Use case was first for the Database Secret Engine, but should be used in future iterations as we don't generally want to show things the user does not
+ * have access to
+ *
+ *
+ * @example
+ * ```js
+ * <SecretListHeaderTab @displayName='Database' @id='database-2' @path='roles' @label='Roles' @tab='roles'/>
+ * ```
+ * @param {string} [displayName] - set on options-for-backend this sets a conditional to see if capabilities are being checked
+ * @param {string} [id] - if fetching capabilities used for making the query url.  It is the name the user has assigned to the instance of the engine.
+ * @param {string} [path] - set on options-for-backend this tells us the specifics of the URL the query should hit.
+ * @param {string} label - The name displayed on the tab.   Set on the options-for-backend.
+ * @param {string} [tab] - The name of the tab.  Set on the options-for-backend.
+ *
+ */
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
+import { task } from 'ember-concurrency';
+
+export default class SecretListHeaderTab extends Component {
+  @service store;
+  @tracked dontShowTab;
+  constructor() {
+    super(...arguments);
+    this.fetchCapabilities.perform();
+  }
+
+  pathQuery(backend, path) {
+    return {
+      id: `${backend}/${path}/`,
+    };
+  }
+
+  @task(function*() {
+    // For now only check capabilities for the Database Secrets Engine
+    // ARG TODO try and think of a way to clean up?  Unsure how.
```

Comment:
Yeah this is pretty gnarly... 🤔 my first thought is maybe utilizing `peekAll` and then using filtering but this is not something for 5pm brain to solve 🙃

Reply:
That might be a good option.  I'll try tackling tmrw morning.  Thanks!